Added .NET 9 (C#) version of benchmark

Braincompiler opened the pull request:
This is the 1:1 translation of the current benchmark code into C# without any optimizations. The benchmark was run on my System (macOS 15.3.2, M1 Max).

Braincompiler committed:
Added .NET 9 (C#) version of benchmark

naveed125 commented on src/benchmark.cs/Program.cs:
```diff
@@ -0,0 +1,98 @@
+      GenerateStrings(STRING_COUNT);
+      stopwatch.Stop();
+
+      return stopwatch.ElapsedMilliseconds;
```
```suggestion
      return Math.Round(stopwatch.Elapsed.TotalMilliseconds, 2);
```

naveed125 replied:
Without this change, the get this data:
```
18	2	2	7.33
12	1	1	4.67
12	1	1	4.67
```

With my change i get more accurate results:

```
17.55	2.24	2.23	7.34
12.28	1.84	1.94	5.35
12.07	1.87	1.88	5.27
```

Braincompiler replied:
Nice, thanks, I haven't found a solution for this immediately. And you have way better benchmark results than me on my system :D

naveed125 reviewed (changes requested):
Thanks for your PR. I made a small suggestion to improve the data. Also, please add the contributors sections of the README file as well to add your name. Thanks.

Braincompiler committed:
Use more precise stopwatch result

Avoid unneeded casting, remove Math.Abs since the random result can never be negative -> slightly better performance
Updated benchmark results
Added myself to the contributors section
